### PR TITLE
Rust edition 2018 + formatting + clippy

### LINKED ISSUE #83: Change edition from `2015` to `2018`
Discovered in the review of #79 .

### PR DESCRIPTION
**Pull request by phip1611**

### CODE REVIEW

**Review by IsaacWoods — APPROVED** [review]

**Comment by IsaacWoods** [comment]
Whoops, approved before I realised that this broke CI. Fine apart from that.

**Comment by phip1611** [comment]
oops, sorry, my bad. I fix it ASAP

**Comment by phip1611** [comment]
Hey @IsaacWoods 
I requested a re-review. Sorry for the additional workload, but I also added code quality improvements reported by clippy. Hope that's okay :)

File: src/header.rs [hunk 1]
```diff
@@ -63,21 +62,6 @@ impl PartialEq<TagType> for TagType {
     }
 }
 
-impl PartialOrd<u32> for TagType {
```
Comment (phip1611) [inline src/header.rs:66]: This was unused and is not required at all. I added this a few PRs earlier by more or less by accident.

File: src/lib.rs [hunk 2]
```diff
@@ -664,85 +666,84 @@ mod tests {
...
-            );
-            assert_eq!(
-                vbe.mode_info.direct_color_attributes,
-                VBEDirectColorAttributes::RESERVED_USABLE
-            );
-            assert_eq!(vbe.mode_info.framebuffer_base_ptr, 4244635648);
-            assert_eq!(vbe.mode_info.offscreen_memory_offset, 0);
-            assert_eq!(vbe.mode_info.offscreen_memory_size, 0);
-        }
+
```
Comment (phip1611) [inline src/lib.rs:669]: on Github Diff this looks huge, but it's just:

`assert_eq!(vbe.mode, 16762);` => `assert_eq!({ vbe.mode }, 16762);`

to fix a future compile error when reading unaligned fields.

File: src/lib.rs [hunk 3]
```diff
@@ -664,85 +666,84 @@ mod tests {
         assert!(bi.vbe_info_tag().is_some());
         let vbe = bi.vbe_info_tag().unwrap();
         use vbe_info::*;
-        unsafe {
```
Comment (phip1611) [inline src/lib.rs:667]: on Github Diff this looks huge, but it's just:

assert_eq!(vbe.mode, 16762); => assert_eq!({ vbe.mode }, 16762);

to fix a future compile error when reading unaligned fields.

Furthermore the unnecessary `unsafe` block was removed.

File: src/lib.rs [hunk 4]
```diff
@@ -322,7 +327,7 @@ impl Reader {
 
     pub(crate) fn read_u8(&mut self) -> u8 {
         self.off += 1;
-        unsafe { core::ptr::read(self.ptr.offset((self.off - 1) as isize)) }
+        unsafe { core::ptr::read(self.ptr.add(self.off - 1)) }
```
Comment (phip1611) [inline src/lib.rs:330]: we don't need to cast `usize` to `isize` this way

File: src/lib.rs (hunk 2 above)
Comment (IsaacWoods) [reply src/lib.rs:669]: Ah cool, thanks for the pointer

**Comment by IsaacWoods** [comment]
> I requested a re-review. Sorry for the additional workload, but I also added code quality improvements reported by clippy. Hope that's okay :)

No worries, that's fine :)

Lots of good cleanup here, and nice to get it to the current edition. Thanks!

**Review by IsaacWoods — APPROVED** [review]

**Pull request merged**